### PR TITLE
Implementation of #157

### PR DESCRIPTION
**Pull request by Waldleufer**
#157

### CODE REVIEW

File: frontend/src/interface/processBackendData.js [hunk 1]
```diff
@@ -67,10 +67,10 @@ export function processBackendData(data, callback) {
...
-    setMaterialCompositionLabels(materialMap.keys());
-    setMaterialCompositionData(materialMap.values());
-    setImpactAssessmentData(impactMap);
-    setColumnChartData(assessmentDataInPercent);
+    setMaterialCompositionLabels(scenarioName, materialMap.keys());
+    setMaterialCompositionData(scenarioName, materialMap.values());
+    setImpactAssessmentData(scenarioName, impactMap);
+    setColumnChartData(scenarioName, assessmentDataInPercent);
 
     callback();
```
Comment (Waldleufer) [inline frontend/src/interface/processBackendData.js:75]: @ScoutAtlas and this is probably about the point where we should store the processed calculation data.

File: frontend/src/interface/projectInterface.js [hunk 2]
```diff
@@ -44,39 +59,63 @@ export function getModels() {
...
+    } else if (scenarioName === scenarioNames.modified) {
+        console.log('compositionData');
+        modifiedMaterialDataInPercent = [];
+        modifiedMaterialCompositionData = Array.from(compositionData);
+        console.log(modifiedMaterialCompositionData);
+        for (let i = 0; i < modifiedMaterialCompositionData.length; i++) {
+            modifiedMaterialDataInPercent[i] = Number(modifiedMaterialCompositionData[i]);
+        }
+        console.log(modifiedMaterialDataInPercent);
     }
```
Comment (Waldleufer) [inline frontend/src/interface/projectInterface.js:92]: When storing the data, we should take this into account 😆 So when scheduling the calculation for the modified scenarioComponent we will have to handle also the storing a little bit different I guess.

File: frontend/src/interface/projectInterface.js [hunk 3]
```diff
@@ -44,39 +59,63 @@ export function getModels() {
...
 /**
  * Getter method to recieve the filtered Material Composititon Data from API
  */
-export function getMaterialCompositionData() {
-    return materialDataInPercent;
+export function getMaterialCompositionData(scenarioName) {
+    if (scenarioName === scenarioNames.baseline) {
+        return materialDataInPercent;
+    } else if (scenarioName === scenarioNames.modified) {
+        return modifiedMaterialDataInPercent;
```
Comment (Waldleufer) [inline frontend/src/interface/projectInterface.js:102]: but I guess this means that they just have to be put to display in a different place.

**Review by Waldleufer — COMMENTED** [review]
Looking very good!

File: frontend/src/components/details/DetailsComponent.js [hunk 4]
```diff
@@ -115,15 +145,32 @@ class DetailsComponent extends Component {
...
+                scenarioNames.baseline,
+                handleFinishedDataRequest
+            );
+            return <LoadingComponent loading />;
+        } else if (this.state.loadingModified) {
+            postCalculationRequest(
+                this.state.secondSelectedScenarioId,
+                scenarioNames.modified,
+                handleFinishedDataRequest
+            );
```
[the remark this answers is not in the document]
Comment (SaiVarunVaranasi) [reply frontend/src/components/details/DetailsComponent.js:180]: Now we will get the entire page as loading but the data for the chosen component only will be updated

File: frontend/src/components/details/PieChartComponent.jsx [hunk 5]
```diff
@@ -9,9 +9,9 @@ import {
...
 
-const PieChartComponent = () => {
-    const series = getMaterialCompositionData();
-    const labels = getMaterialCompositionLabels();
+const PieChartComponent = (props) => {
+    const series = getMaterialCompositionData(props.scenarioName);
+    const labels = getMaterialCompositionLabels(props.scenarioName);
 
     console.log(series);
     console.log(labels);
```
[the remark this answers is not in the document]
Comment (SaiVarunVaranasi) [reply frontend/src/components/details/PieChartComponent.jsx:18]: Done

File: frontend/src/components/details/MobileTableComponent.jsx [hunk 6]
```diff
@@ -13,7 +13,7 @@ import PropTypes from 'prop-types';
 class MobileTableComponent extends Component {
     state = {
         headers: getImpactCategoriesTableHeaders(),
-        rows: getImpactAssessmentData()
+        rows: getImpactAssessmentData(this.props.scenarioName)
```
[the remark this answers is not in the document]
Comment (SaiVarunVaranasi) [reply frontend/src/components/details/MobileTableComponent.jsx:16]: Done

File: frontend/src/components/details/SelectScenarioComponent.jsx [hunk 7]
```diff
@@ -19,7 +20,9 @@ class SelectScenarioComponent extends Component {
      * @param item: selected scenario
      */
     onDropDownItemSelectedHandler = (item) => {
-        this.props.newScenarioHandler(item);
+        if (this.props.scenarioName === scenarioNames.modified)
```
[the remark this answers is not in the document]
Comment (SaiVarunVaranasi) [reply frontend/src/components/details/SelectScenarioComponent.jsx:22]: Done

File: frontend/src/components/details/TableComponent.jsx [hunk 8]
```diff
@@ -11,7 +11,7 @@ import {
 class TableComponent extends Component {
     state = {
         headers: getImpactCategoriesTableHeaders(),
-        rows: getImpactAssessmentData()
+        rows: getImpactAssessmentData(this.props.scenarioName)
```
[the remark this answers is not in the document]
Comment (SaiVarunVaranasi) [reply frontend/src/components/details/TableComponent.jsx:15]: Done

File: frontend/src/interface/projectInterface.js (hunk 3 above)
Comment (SaiVarunVaranasi) [reply frontend/src/interface/projectInterface.js:102]: yes. For now, the data is stored in two different sets of variables for Baseline and Modified scenarios. This data is called based on the scenario, if it is Modified scenario/Baseline Scenario.

**Review by Waldleufer — COMMENTED** [review]
Thank you!

File: frontend/src/components/details/DetailsComponent.js [hunk 9]
```diff
@@ -41,6 +50,8 @@ class DetailsComponent extends Component {
         super(props);
         this.state.selectedScenarioType = props.selectedProduct.scenarioType;
         this.state.selectedScenarioId = props.selectedProduct.productID;
+        this.state.secondSelectedScenarioType = props.selectedProduct.scenarioType;
+        this.state.secondSelectedScenarioId = props.selectedProduct.productID;
```
[the remark this answers is not in the document]
Comment (gandompm) [reply frontend/src/components/details/DetailsComponent.js:59]: thank you :)

File: frontend/src/components/details/DetailsComponent.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (gandompm) [reply frontend/src/components/details/DetailsComponent.js]: done

**Review by Waldleufer — COMMENTED** [review]
lgtm

**Pull request merged**